lib/ical_filter_proxy/filter_rule.rb

```diff
@@ -27,6 +27,10 @@ def evaluate(event_data, value)
           event_data == value
         when 'startswith'
           event_data.start_with?(value)
+        when 'includes'
+          event_data.include?(value)
+        when 'matches'
+          event_data =~ eval(value)
```

Comment:
`eval` is a pretty dangerous method to be using here. Instead use `Regexp.new` to compile your regular expression.

Reply:
Makes sense, will do and update

Reply:
The only good thing about eval was that I could add /<pattern>/<options>.
With Regex.new I have do pass <pattern> and the options might be "hard coded".
You see any chance or have any preference?

Reply:
Updated with a proposal, please have a look @darkphnx

Reply:
It's not very pretty, but you could parse a // expression yourself and set options in Regexp.new. Something like:

```ruby
REGEXP_OPTS_LOOKUP = { 
  'i' => Regexp::IGNORECASE, 
  'm' => Regexp::MULTILINE, 
  'x' => Regexp::EXTENDED,
}

def match_value_to_regexp(value) 
  parsed_value = /^\/(?<pattern>.+)\/(?<opts>.*)$/.match(value)

  opts = parsed_value[:opts]
    .chars
    .map { |opt| REGEXP_OPTS_LOOKUP[opt] }
    .compact
    .inject { |memo, opt| memo | opt }

  Regexp.new(parsed_value[:pattern], opts)
end

value = "/my-search/i"
regexp = match_value_to_regexp(value)
# => /my-search/i
```